File: 01-1.rb
#!/usr/bin/env ruby

lines = File.read('01.input').lines.map(&:strip)

print(lines.map do |line|
  numerals = line.chars.reject { |c| c.match('[[:alpha:]]') }
  (numerals[0] + numerals[-1]).to_i
end.sum, "\n")

File: 01-1.py
#!/usr/bin/env python

lines = [row.strip() for row in open('01.input').readlines()]

sum = 0

for line in lines:
    numerals = [c for c in line if c.isdigit()]
    sum += int(numerals[0] + numerals[-1])

print(sum)
